rafthttp/pipeline.go

```diff
@@ -128,31 +132,43 @@ func (p *pipeline) post(data []byte) (err error) {
 		}
 	}()
 
-	resp, err := p.tr.pipelineRt.RoundTrip(req)
-	done <- struct{}{}
-	if err != nil {
-		p.picker.unreachable(u)
-		return err
-	}
-	b, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		p.picker.unreachable(u)
-		return err
-	}
-	resp.Body.Close()
-
-	err = checkPostResponse(resp, b, req, p.peerID)
-	if err != nil {
-		p.picker.unreachable(u)
-		// errMemberRemoved is a critical error since a removed member should
-		// always be stopped. So we use reportCriticalError to report it to errorc.
-		if err == errMemberRemoved {
-			reportCriticalError(err, p.errorc)
+	go func() {
```

Comment:
@xiang90 i tested it with old code, it will not work. The test case fails. So this code is required for it to work properly in this case.
![image](https://cloud.githubusercontent.com/assets/20169156/22477119/e4733ccc-e80a-11e6-97bc-b73139f168f4.png)